ECoG_MAE/utils.py

```diff
@@ -16,6 +17,44 @@ def seed_everything(seed=0, cudnn_deterministic=True):
     torch.cuda.manual_seed_all(seed)
 
 
+
+def resample_mean_signals(signal: np.array, old_fs: int, new_fs: int) -> np.array:
+    """Resample signal with sampling rate of old_fs Hz to new_fs Hz by taking means over windows of data.
+
+    Args:
+        signal (np.array): shape [bands, num_electrodes, samples]
+        old_fs (int): Old sample rate in Hz.
+        new_fs (int): Sample rate to resample to in Hz.
+        
+    Returns:
+        np.array: Resampled signal with the new sample rate.
+    """
+    window_width = old_fs / new_fs
+    num_samples = signal.shape[2]
+    # TODO: revisit using ceil here. By using ceil our final entry in our new array may be averaged
```

Comment:
Using np.ceil looks reasonable to me - the other option is to just truncate the signal if ((num_samples * new_fs) % old_fs) != 0